server.go

```diff
@@ -446,32 +1234,30 @@ func (s *server) handleQuery(querymsg interface{}, state *peerState) {
 			// however it is statistics for purely informational purposes
 			// and we don't really care if they are raced to get the new
 			// version.
-			p.StatsMtx.Lock()
 			info := &btcjson.GetPeerInfoResult{
-				ID:             p.id,
-				Addr:           p.addr,
-				Services:       fmt.Sprintf("%08d", p.services),
-				LastSend:       p.lastSend.Unix(),
-				LastRecv:       p.lastRecv.Unix(),
-				BytesSent:      p.bytesSent,
-				BytesRecv:      p.bytesReceived,
-				ConnTime:       p.timeConnected.Unix(),
-				TimeOffset:     p.timeOffset,
-				Version:        p.protocolVersion,
-				SubVer:         p.userAgent,
-				Inbound:        p.inbound,
-				StartingHeight: p.startingHeight,
-				CurrentHeight:  p.lastBlock,
+				ID:             p.ID(),
```

Comment:
Having to grab the mutex for each individual field here is not ideal.  I'd suggest a method on the peer like:

```Go
// StatsSnapshot returns a snapshot of the current peer statistics.
func (p *Peer) StatsSnapshot() *StatsSnap {
	p.statsMtx.RLock()
	defer p.statsMtx.RUnlock()

	// Copy the relevant fields into the snapshot...
	return &StatsSnap{
		ID: p.id,
		Addr: p.addr,
		// ...
	}
}
```

Then you could simply do here:

```Go
statsSnap := p.StatsSnapshot()
info := &btcjson.GetPeerInfoResult{
	ID: statsSnap.ID,
	// ...
	SyncNode: p == syncPeer,
}
```